### ISSUE TITLE
Feature Request: Diary Automatic daywise ready todo list.

### ISSUE THREAD

**Issue by eyevish** [vimwiki/vimwiki#1028]
Disclaimer:  _I am new to Vim configurations and vim scripts. I am new to using github too so request you to cut some slack on any misconduct_ :-) .
**Feature request : as a user when I create a DiaryNote the page should be prefilled with some ToDo list that I have set for each day of the week.**

My initial requirement was to use the diary mainly for the TODO task which quite often are repeating weekly. But keying the todo every time or even copy paste option seemed a little tedious. 
So here is what I did as my hacked automatic version of the feature I wanted. I create  a wikiPage for each day of the week as WDAY_TODO.md, where WDAY will be like TUE for Tuesday. 
In the VIMRC I add a small function as below that is called instead of VimwikiMakeDiaryNote : 
```
function! NewDiaryNote()
	let WeekDays = ["","Mon","Tue","Wed","Thu","Fri","Sat","Sun"]
	let l:dayName= WeekDays[strftime("%w")]
	let l:fname = expand("~/documents/vimwiki/" . WeekDays[strftime("%w")] . "_todo.md")
	VimwikiMakeDiaryNote
	if line('$')==1
	if filereadable(fname)
		execute "read" fname
	endif
	call deletebufline("%",1)
	endif
endfunction
```
This is my first attempt at VIM script so surely #_errors > #Lines is a big possibility. I am hoping some may make this a inbuilt function in this Already SUPER Awesome plug.
A small issue if I may include here is that when I use Windows OS neovim and Markdown as syntax the VimWikiDiaryGenerateLinks did not work. Emptied what was already in the index page. 
To workaround the problem I change the line 1262 in base.vim from (as I said, I am still learning the ropes when it comes to vim scripts.)
```if start_lnum == line('$')```
to 
``` if start_lnum == line('$') && 1==2 ```

Once again thanks for this wonderful plugin.

**Comment by tinmarino** [comment]
`:h vimwiki-option-diary_frequency` ?

**Comment by eyevish** [comment]
> 
> 
> `:h vimwiki-option-diary_frequency` ?

I am not sure I understand you tinmarino . Am I missing something that is already implemented ? For me the 
''':h vimwiki-option-diary_frequency'''
returned error of "No help for ..."

**Comment by tinmarino** [comment]
Read [this](https://github.com/vimwiki/vimwiki/blob/a6878015e6180532d7e5eb13266a5810742819c5/doc/vimwiki.txt#L2522) help chapter in foc/vimwiki refering to the diary frecuency. "diary" can be weekly, monthly or yearly changing the `diary_frecuency` option.

**Comment by eyevish** [comment]
OK - I now understand that I have not been good with explaining the feature requirement.  - 
My query was not how to control the frequency of the diary, but it is good to learn from you about this feature.
Let me try again in simpler words.

> when I do <leader>w<leader>w opens the plugin opens an empty diary page today. 
instead it would be nice if it opens a page that is prefilled with some To Do's that are preset by the user. for example every Saturday there is a to-do that get prefilled as a task/note to **- [ ] get grocery**
In my case (using the code from the first post), I extract it from a wiki page, and initialize the first line with the day and date.
----
Hope it was able to clear the confusion. the crappy code above works for me, but I was hoping to have a more professional implementation.  
Also in vimrc how to set **<leader>x<leader>x map** is it possible? I have to use <leader>x<space>x where space is my leader to make it work.

**Comment by tinmarino** [comment]
> opens a page that is prefilled with some

OK, this is not implemented yet, not a bad idea.

> Also in vimrc how to set *xx map* is it possible?

Please try to be as specific as possible, even if you need 20 lines to
explain it.
Generic answer relative to map: `:map <leader>a :echom "toto"<CR>`

**Comment by tinmarino** [comment]
> The crappy code above works for me, but I was hoping to have a more professional implementation.

I love cleaning code (much easyer than writing from scratch which needs inspiration).
Give me one month and I'll send you a new command on this thread.

**Comment by eyevish** [comment]
Thanks I look forward to it and love the plugin.
the Map question should not have been here, but it was a casual question but the markdown ruined it. My bad I did not preview it. My question was on how to map leader+x+leader+x  kind of comb. in vimrc. but this is more of a vim question, please ignore it.

**Comment by ahillio** [comment]
I've got a python script that I use as a template for generating my daily journal... it gives me a boilerplate for stuff like dreams/challenges/accomplishments/gratitudes/etc and TODOs... which includes a daily checklist and a few "smart" sections:

1. Query taskwarrior database (using tasklib python library) for tasks that are due before tomorrow
2. Query khal calendar to get any scheduled appointments etc
3. Anything from yesterday's "tomorrow" section

I don't know how useful this would be to others, but I just stumbled across this issue while looking for something else, so figured I'd share...

` autocmd BufNewFile ~/Documents/wiki/diary/[0-9]\\\{4\}-[0-9]\\\{2\}-[0-9]\\\{2\}.mkd :execute 'silent 0r !vimwiki-diary-template.py' | normal 7gg` uses `vimwiki-diary-template.py` to create the new diary page and then moves the cursor to line#7 (because the first 6 lines have miscellaneous stuff, line7 is where I start typing).

An abbreviation of the script looks like:

```
#!/usr/bin/python3
import os
import sys
import subprocess
from datetime import datetime, timedelta
from tasklib import TaskWarrior

# Date
date = datetime.strftime(datetime.now(), "%A %d %B %Y")
uline = '--'
length = len(date)
i = 0
while i < length:
    uline += '-'
    i += 1

# Today's calendar items
calEvents = ''
calEvents = subprocess.run("khal list today 1d | tail -n +2 | sed 's/^/- [ ] /'", shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True)
calEvents = calEvents.stdout + '\n'

# Tasks
tw = TaskWarrior('/home/alec/.task')
tasksDue = tw.tasks.filter(status='pending', due__before='tomorrow')
todaysTasks = ''
for task in tasksDue:
    todaysTasks += '* [ ] ' + task['description'] + '  #' + task['uuid'][:8] + '\n'

# Notes from yesterday's diary
yesNotes = ''
yesJournal = '/home/alec/Documents/wiki/diary/' + datetime.strftime(datetime.now() - timedelta(1), '%Y-%m-%d') + '.mkd'
# note this will break the template script if a journal entry for yesterday does not exist
with open (yesJournal, "r") as f:
    for line in f:
        if "## Tomorrow" in line:
            for line in f:
                #line = ' ' + line
                yesNotes = yesNotes + line

# BEGIN TEMPLATE
template = """# {date}
{uline}

<misc stuff>

## TODO
- [ ] get up
- [ ] make bed
- [ ] etc
# now include our cool scripted stuff:
{yesNotes}{calEvents}{todaysTasks}

<more misc stuff>
## Tomorrow"""

print(template.format(uline=uline, date=date, todaysTasks=todaysTasks, yesNotes=yesNotes, calEvents=calEvents))
```

Thus my daily todo list has standard things to do, as well as three different kind of "scheduled" things to do.

**Comment by lyda** [comment]
I use [vim-template](https://github.com/aperezdc/vim-template) for templates.  I added a template file in `~/.vim/templates/` (can't recall if that's the standard path) with the rather annoying name of `=template=[0-9][0-9][0-9][0-9]-[0-9][0-9]-[0-9][0-9].md`. A new diary entry is loaded with that.

Not sure if you can lock it down just to the diary dir - I don't have date files like that elsewhere so this should be fine.
